src/services/kratos/auth-device-account.ts

```diff
@@ -0,0 +1,249 @@
+import {
+  LikelyNoUserWithThisPhoneExistError,
+  LikelyUserAlreadyExistError,
+} from "@domain/authentication/errors"
+import { CreateIdentityBody, Identity, UpdateIdentityBody } from "@ory/client"
+import * as jose from "node-jose"
+
+import {
+  getAppCheckConfig,
+  getDefaultAccountsConfig,
+  getJwksArgs,
+  getKratosPasswords,
+  isCI,
+  isDev,
+} from "@config"
+
+import jwksRsa from "jwks-rsa"
+import jsonwebtoken from "jsonwebtoken"
+
+import { AccountLevel } from "@domain/accounts"
+import { createAccountForDeviceAccount } from "@app/accounts"
+
+import {
+  AuthenticationKratosError,
+  IncompatibleSchemaUpgradeError,
+  KratosError,
+  UnknownKratosError,
+} from "./errors"
+
+import { kratosAdmin, kratosPublic, toDomainIdentityPhone } from "./private"
+
+export const AuthWithDeviceAccountService = () => {
+  const password = getKratosPasswords().masterUserPassword
+
+  const loginDeviceAccount = async ({
+    deviceId,
+  }: {
+    deviceId: DeviceId
+  }): Promise<LoginWithDeviceSchemaResponse | KratosError> => {
+    const identifier = deviceId
+    const method = "password"
+
+    try {
+      const flow = await kratosPublic.createNativeLoginFlow()
+      const result = await kratosPublic.updateLoginFlow({
+        flow: flow.data.id,
+        updateLoginFlowBody: {
+          identifier,
+          method,
+          password,
+          traits: {
+            device_id: identifier,
+          },
+        },
+      })
+      const sessionToken = result.data.session_token as SessionToken
+
+      // note: this only works when whoami: required_aal = aal1
+      const kratosUserId = result.data.session.identity.id as UserId
+
+      return { sessionToken, kratosUserId }
+    } catch (err) {
+      if (err.message === "Request failed with status code 400") {
+        return new LikelyNoUserWithThisPhoneExistError(err.message || err)
+      }
+
+      if (err.message === "Request failed with status code 401") {
+        return new AuthenticationKratosError(err.message || err)
+      }
+
+      return new UnknownKratosError(err.message || err)
+    }
+  }
+
+  const createDeviceIdentity = async ({
+    deviceId,
+  }: {
+    deviceId: DeviceId
+  }): Promise<CreateKratosUserForDeviceSchemaResponse | KratosError> => {
+    const traits = { device: deviceId }
```

Comment:
```suggestion
    const traits = { deviceId: deviceId }
```

I would also be consistent at this layer, ie: using `deviceId` instead of `device`.